pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1796,16 +1796,16 @@ private boolean updateExistedSegment(String tableNameWithType, String segmentNam
     }
 
     // Set all partitions to online so that they load the new segment data
-    do {
-      final IdealState idealState = _helixAdmin.getResourceIdealState(_helixClusterName, tableNameWithType);
+    HelixHelper.updateIdealState(_helixZkManager, tableNameWithType, idealState -> {
```

Comment:
I thought this is always done using CAS where you try to update the version and if the version update fails, it implies someone beat you while updating the ideal state concurrently. In that case, you fetch the updated ideal state again and start. Were we not following this protocol?

Reply:
At least the TableRebalancer code follows this.

Reply:
It is following the this protocol in `updateIdealState` method. You can check the code here:
https://github.com/apache/incubator-pinot/blob/00fcb1d27f8daec48c1d87d46abfc8b60cfb32cf/pinot-common/src/main/java/org/apache/pinot/common/utils/helix/HelixHelper.java#L120